x/nft/internal/types/msgs.go

```diff
@@ -0,0 +1,240 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+/* --------------------------------------------------------------------------- */
+// MsgTransferNFT
+/* --------------------------------------------------------------------------- */
+
+// MsgTransferNFT defines a TransferNFT message
+type MsgTransferNFT struct {
+	Sender    sdk.AccAddress
+	Recipient sdk.AccAddress
+	Denom     string
+	ID        string
+}
+
+// NewMsgTransferNFT is a constructor function for MsgSetName
+func NewMsgTransferNFT(sender, recipient sdk.AccAddress, denom, id string) MsgTransferNFT {
+	return MsgTransferNFT{
+		Sender:    sender,
+		Recipient: recipient,
+		Denom:     strings.TrimSpace(denom),
+		ID:        strings.TrimSpace(id),
+	}
+}
+
+// Route Implements Msg
+func (msg MsgTransferNFT) Route() string { return RouterKey }
+
+// Type Implements Msg
+func (msg MsgTransferNFT) Type() string { return "transfer_nft" }
+
+// ValidateBasic Implements Msg.
+func (msg MsgTransferNFT) ValidateBasic() sdk.Error {
+	if msg.Denom == "" {
+		return ErrInvalidCollection(DefaultCodespace)
+	}
+	if msg.Sender.Empty() {
+		return sdk.ErrInvalidAddress("invalid sender address")
+	}
+	if msg.Recipient.Empty() {
+		return sdk.ErrInvalidAddress("invalid recipient address")
+	}
+	if msg.ID == "" {
+		return ErrInvalidNFT(DefaultCodespace)
+	}
+
+	return nil
+}
+
+// GetSignBytes Implements Msg.
+func (msg MsgTransferNFT) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners Implements Msg.
+func (msg MsgTransferNFT) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+/* --------------------------------------------------------------------------- */
+// MsgEditNFTMetadata
+/* --------------------------------------------------------------------------- */
+
+// MsgEditNFTMetadata edits an NFT's metadata
+type MsgEditNFTMetadata struct {
```

Comment:
The ID, and Denom are meant to be permanent and used to reference the token. Name should be considered part or the editable metadata. Happy to hear your thoughts if you disagree though. I'm also interested in moving metadata out into it's own module where it can be nested and expandable for as wide a variety of use cases as possible.